File: movies_repository/src/main/java/com/nikita/popularmovies/common/models/MovieDetails.java
package com.nikita.popularmovies.common.models;

import java.util.Collections;
import java.util.List;

public final class MovieDetails {
  public MoviePreview moviePreview;
  public List<Video> videos;
  public List<Review> reviews;
  public boolean isSaved = false;

  public MovieDetails() {}

  public MovieDetails(MoviePreview moviePreview) {
    this.moviePreview = moviePreview;
    videos = Collections.emptyList();
    reviews = Collections.emptyList();
  }

  public MovieDetails(MoviePreview moviePreview, List<Video> videos, List<Review> reviews) {
    this.moviePreview = moviePreview;
    this.videos = videos;
    this.reviews = reviews;
  }

  public MoviePreview getMoviePreview() {
    return moviePreview;
  }

  public List<Video> getVideos() {
    return videos;
  }

  public List<Review> getReviews() {
    return reviews;
  }
}

File: movies_repository/src/main/kotlin/com/nikita/popularmovies/common/models/MovieDetails.kt
package com.nikita.popularmovies.common.models

import android.arch.persistence.room.*

// Room cant work with kotlin constructors
// default parameters need to use constructor without params in room

@Entity(foreignKeys = arrayOf(ForeignKey(entity = MoviePreview::class, parentColumns = arrayOf("id"), childColumns = arrayOf("movie_id"))),
  indices = arrayOf(Index(value = "movie_id")))
data class Review(@field:PrimaryKey val id: String,
                  @field:ColumnInfo(name = "movie_id") val movieId: String,
                  val author: String,
                  val content: String,
                  val url: String)

@Entity(foreignKeys = arrayOf(ForeignKey(entity = MoviePreview::class, parentColumns = arrayOf("id"), childColumns = arrayOf("movie_id"))),
  indices = arrayOf(Index(value = "movie_id")))
data class Video(@field:PrimaryKey val id: String,
                 @field:ColumnInfo(name = "movie_id") val movieId: String,
                 val key: String,
                 val name: String,
                 val site: String,
                 val type: String)